code/game/objects/effects/landmarks.dm

```diff
@@ -1,95 +1,64 @@
+var/list/landmarks_list = list() //list of all landmarks
+var/list/landmarks_by_type = list() //list of landmark types associated with turf lists
+
+//Returns a list of all landmark turfs or an empty list
+//If readonly is false, it returns a COPY of the list, which can be changed without affecting the original list
+/proc/get_landmarks(input_type, readonly = TRUE)
+	var/list/L = landmarks_by_type[input_type] || list()
+	if(!readonly)
+		L = L.Copy()
+
+	return L
+
+/proc/pick_landmark(input_type, backup_type = /obj/effect/landmark/latejoin)
+	var/list/L = get_landmarks(input_type)
+	if(!L.len && (backup_type != input_type))
+		L = get_landmarks(backup_type)
+	if(!L.len)
+		return null
+
+	return pick(L)
+
 /obj/effect/landmark
 	name = "landmark"
 	icon = 'icons/mob/screen1.dmi'
 	icon_state = "x2"
 	anchored = 1
 	w_type=NOT_RECYCLABLE
 
+	var/destroy_on_creation = FALSE
```

Comment:
I fail to see the advantage of these destroy on creation ones. All I can see it do is make adminbus harder.